library/Makefile

```diff
@@ -1,8 +1,21 @@
 
 # Also see "include/mbedtls/config.h"
 
+ifeq ($(CC),iccarm)
+CFLAGS	?= --arm -Om --silent
+
+# gcc comparable warning flags are not present in IAR.
+# All warnings are enabled by default.
+WARNING_CFLAGS ?=
+else
 CFLAGS	?= -O2
 WARNING_CFLAGS ?=  -Wall -W -Wdeclaration-after-statement
+endif
+
+# IAR archive command does not have ar compatible options 
+ifneq ($(AR),iarchive)
```

Comment:
Theoretically yes but mostly we add tools to the path. So it should not be an issue.
We can take alternative approach of setting environment variables  for everything set inside ```if``` blocks. But then there would be lot of environment variables to set.
Conditionally setting variables inside make does make it bit versatile.